cs/test/expressions/SerializeCB.expressions

```diff
@@ -1199,6 +1199,72 @@
         };
+        .Block() {
+            .Call $writer.WriteFieldBegin(
+                .Constant<Bond.BondDataType>(BT_STRUCT),
+                .Constant<System.UInt16>(85),
+                .Constant<Bond.Metadata>(_dt));
+            .Block(System.Object $obj) {
+                $obj = (System.Object)$Example_obj._dt;
+                .Block(ExpressionsTest.Generic`1[System.DateTime] $Generic`1_obj) {
+                    $Generic`1_obj = (ExpressionsTest.Generic`1[System.DateTime])$obj;
+                    .Call $writer.WriteStructBegin(.Constant<Bond.Metadata>(Generic<int64>));
+                    .If ($Generic`1_obj.value != .Call ExpressionsTest.BondTypeAliasConverter.Convert(
+                        0L,
+                        .Default(System.DateTime))) {
+                        .Block() {
+                            .Call $writer.WriteFieldBegin(
+                                .Constant<Bond.BondDataType>(BT_INT64),
+                                .Constant<System.UInt16>(0),
+                                .Constant<Bond.Metadata>(value));
+                            .Call $writer.WriteInt64(.Call ExpressionsTest.BondTypeAliasConverter.Convert(
+                                    $Generic`1_obj.value,
+                                    .Default(System.Int64)));
+                            .Call $writer.WriteFieldEnd()
+                        }
+                    } .Else {
+                        .Call $writer.WriteFieldOmitted(
+                            .Constant<Bond.BondDataType>(BT_INT64),
+                            .Constant<System.UInt16>(0),
+                            .Constant<Bond.Metadata>(value))
+                    };
+                    .Call $writer.WriteStructEnd()
+                }
+            };
+            .Call $writer.WriteFieldEnd()
+        };
+        .Block() {
+            .Call $writer.WriteFieldBegin(
+                .Constant<Bond.BondDataType>(BT_STRUCT),
+                .Constant<System.UInt16>(86),
+                .Constant<Bond.Metadata>(_dt2));
+            .Block(System.Object $obj) {
+                $obj = (System.Object)$Example_obj._dt2;
+                .Block(ExpressionsTest.GenericNothing`1[System.DateTime] $GenericNothing`1_obj) {
+                    $GenericNothing`1_obj = (ExpressionsTest.GenericNothing`1[System.DateTime])$obj;
+                    .Call $writer.WriteStructBegin(.Constant<Bond.Metadata>(GenericNothing<int64>));
+                    .If ($GenericNothing`1_obj.value != null) {
+                        .Block() {
+                            .Call $writer.WriteFieldBegin(
+                                .Constant<Bond.BondDataType>(BT_INT64),
+                                .Constant<System.UInt16>(0),
+                                .Constant<Bond.Metadata>(value));
+                            .Call $writer.WriteInt64((System.Int64).Call ExpressionsTest.BondTypeAliasConverter.Convert(
```

Comment:
It looks like we can avoid calling the `Convert` for `T?` (an extra overload) and delegate to existing version that expect `T` since the above check for `null`?